### PR TITLE
Set span in ProcessEventDelivery

### PR DESCRIPTION
**Pull request by ogbanugot**
- Sets a span in ProcessEventDelivery using newrelic tracer, so that we can have an estimate of the average time it takes to process an event delivery.

### CODE REVIEW

**Review by jirevwe — APPROVED** [review]

File: worker/task/process_event_delivery.go [hunk 1]
```diff
@@ -43,8 +45,14 @@ type SignatureValues struct {
...
 }
 
-func ProcessEventDelivery(appRepo datastore.ApplicationRepository, eventDeliveryRepo datastore.EventDeliveryRepository, groupRepo datastore.GroupRepository, rateLimiter limiter.RateLimiter) func(*queue.Job) error {
+func ProcessEventDelivery(appRepo datastore.ApplicationRepository, eventDeliveryRepo datastore.EventDeliveryRepository, groupRepo datastore.GroupRepository, rateLimiter limiter.RateLimiter, tracer tracer.Tracer) func(*queue.Job) error {
 	return func(job *queue.Job) error {
+		var txn *newrelic.Transaction
+		if tracer != nil {
+			txn = tracer.StartTransaction("ProcessEventDelivery")
+		}
+		defer txn.End()
```
Comment (danvixent) [inline worker/task/process_event_delivery.go:54]: Should this guy also be in the if condition?

**Comment by jirevwe** [comment]
@ogbanugot do you think we need to add a span for `ProcessEventCreation()` too?

**Comment by danvixent** [comment]
> @ogbanugot do you think we need to add a span for ProcessEventCreation() too?

I personally think we should @jirevwe

File: worker/task/process_event_delivery.go (hunk 1 above)
Comment (ogbanugot) [reply worker/task/process_event_delivery.go:54]: No, if we put it in the condition won't the transaction be scoped to the if condition?

Comment (danvixent) [reply worker/task/process_event_delivery.go:54]: Yes it would, because if the condition is false, txn will be `nil` and calling txn.End will panic

**Comment by jirevwe** [comment]
> > @ogbanugot do you think we need to add a span for ProcessEventCreation() too?
> 
> I personally think we should @jirevwe

I agree, though it's out of scope of this PR. Plus I think we have all the data we need in the event delivery record, hmm.

**Review by danvixent — APPROVED** [review]

**Pull request merged**